src/algorithms/signal_source/gnuradio_blocks/fifo_reader.cc

```diff
@@ -30,7 +33,10 @@ FifoReader::FifoReader(const std::string &file_name, const std::string &sample_t
           gr::io_signature::make(0, 0, 0),                    // no input
           gr::io_signature::make(1, 1, sizeof(gr_complex))),  // <+MIN_OUT+>, <+MAX_OUT+>, sizeof(<+OTYPE+>)
       file_name_(file_name),
-      sample_type_(sample_type)
+      sample_type_(sample_type),
+      fifo_buffer_(new char[FIFO_SIZE]),
```

Comment:
Just as a rule of thumb, if you are calling `new` explicitly, you've missed the modern C++ way. Either containers or `make_unique<T>`/`make_shared<T>` should be preferred.

Using RAII constructs has the (intentional) benefit of relieving you of destructor responsibilities.

Reply:
Obsolete after [77c682e](https://github.com/gnss-sdr/gnss-sdr/pull/584/commits/77c682e8b312c645d77e60289b0d67b33e67c633).